pkg/newstorage/newstorage.go

```
@@ -84,9 +94,9 @@ type Store interface {
 	GetTags(k string) ([]Tag, error)
 
 	// GetBulk fetches the values associated with the given keys.
-	// If a key doesn't exist, then a nil []byte is returned for that value and is not considered an error.
+	// If a key doesn't exist, then a nil []byte is returned for that value. It is not considered an error.
 	// Depending on the implementation, this method may be faster than calling Get for each key individually.
-	GetBulk(keys ...string) ([][]byte, error)
```

You should return a generic error type from functions.